XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Simulation/SimulatedControllerDataProvider.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Copyright (c) XRTK. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+using XRTK.Definitions.Controllers.Simulation;
+using XRTK.Definitions.Devices;
+using XRTK.Definitions.Utilities;
+using XRTK.Interfaces.InputSystem;
+using XRTK.Interfaces.Providers.Controllers;
+using XRTK.Services;
+
+namespace XRTK.Providers.Controllers.Simulation
+{
+    /// <summary>
+    /// The simulated controller data provider is mainly responsible for managing
+    /// active simulated controllers, such as hand controllers. All active controllers
+    /// at a given time are accessible via <see cref="BaseControllerDataProvider.ActiveControllers"/>.
+    /// </summary>
+    public class SimulatedControllerDataProvider : BaseControllerDataProvider
+    {
+        private SimulationTimeStampStopWatch simulatedUpdateStopWatch;
+        private long lastSimulatedUpdateTimeStamp = 0;
+
+        private bool leftControllerIsAlwaysVisible = false;
+        private bool rightControllerIsAlwaysVisible = false;
+        private bool leftControllerIsTracked = false;
+        private bool rightControllerIsTracked = false;
+
+        /// <summary>
+        /// Gets the active profile for the simulated controller data provider.
+        /// </summary>
+        public SimulatedControllerDataProviderProfile Profile { get; }
```

Comment:
Profiles should never be exposed outside this class.

Please mirror the properties you want to expose and add them to the services interface.

Reply:
The controller needs the profile reference, because it contains the configured recorded hand pose s to simulated.

Reply:
See prev comment about this. We cannot expose profiles outside of the services.

Reply:
See prev comment about this. We cannot expose profiles outside the services they live in.